src/Sparrow/ByteString.cs

```diff
@@ -725,10 +796,19 @@ public void Release(ref ByteString value)
 
         private SegmentInformation AllocateSegment(int size)
```

Comment:
There is some code duplication between `AllocateSegment` and `AllocateExternalSegment` that should probably be merged